src/System.Net.WebClient/tests/WebClientTest.cs

```diff
@@ -54,23 +53,29 @@ public static void DownloadData_InvalidArguments_ThrowExceptions()
             AssertExtensions.Throws<ArgumentNullException>("address", () => { wc.DownloadDataAsync((Uri)null); });
             AssertExtensions.Throws<ArgumentNullException>("address", () => { wc.DownloadDataAsync((Uri)null, null); });
 
-            AssertExtensions.Throws<ArgumentNullException>("address", () => { wc.DownloadDataTaskAsync((string)null); });
+            if (!PlatformDetection.IsFullFramework) // netfx has bug and returns wrong argument name in exception.
+            {
+                AssertExtensions.Throws<ArgumentNullException>("address", () => { wc.DownloadDataTaskAsync((string)null); });
```

Comment:
The "wrong" parameter name is happening because on .NET Framework, the parameter check isn't done at the public API level.  It is being done in a lower layer Uri method and that method has a parameter name of "path".  So, the argument exception is bubbling up from that method which is sort of wrong.  We shouldn't be leaking that argument exception (and parameter name) since it is an implementation detail.  But that is a NETFX bug.